### PR TITLE
Add Woo button to plugins list

### PR DESCRIPTION
**Pull request by bgrgicak**
### Changes proposed in this Pull Request:

We want to improve the discoverability of Woo themes and extensions to help merchants customize their stores.
To do this we would like to add a Woo button to the Plugins list

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the [plugins list](http://localhost:8888/wp-admin/plugin-install.php)
2. Confirm that there is a Woo tab
3. Click on the tab
4. It should redirect you to the in-app Marketplace Extensions tab
5. Confirm that the url includes `&ref=plugins`
6. Click on a product card
7. The Woo.com url should include `&utm_content=plugins` at the end

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

Screenshot 2023-11-22 at 12 30 36

### CODE REVIEW

**Comment by bgrgicak** [comment]
The CI errors are inherited from the feature branch.

File: plugins/woocommerce/src/Internal/Admin/Marketplace.php [hunk 1]
```diff
@@ -71,4 +78,59 @@ public function enqueue_scripts( $hook_suffix ) {
...
 		wp_enqueue_script( 'updates' );
 	}
+
+	/**
+	 * Add a Woo Marketplace link to the plugin install action links.
+	 * @param array $tabs Plugins list tabs.
+	 * @return array
+	 */
+	public function add_woo_plugin_install_action_link( $tabs ) {
+		$tabs[self::MARKETPLACE_TAB_SLUG] = _x( 'Woo', 'Plugin Installer', 'wccom' );
```
Comment (andfinally) [inline plugins/woocommerce/src/Internal/Admin/Marketplace.php]: Does the brand name need to be translated?

File: plugins/woocommerce/src/Internal/Admin/Marketplace.php [hunk 2]
```diff
@@ -71,4 +78,59 @@ public function enqueue_scripts( $hook_suffix ) {
...
+	 * Add styles to the plugin install page.
+	 */
+	public function add_plugins_page_styles() {
+		?>
+		<style>
+			.plugin-install-woo > a::after {
+				content: "";
+				display: inline-block;
+				background-image: url("data:image/svg+xml,%3Csvg width='16' height='16' viewBox='0 0 16 16' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill-rule='evenodd' clip-rule='evenodd' d='M8.33321 3H12.9999V7.66667H11.9999V4.70711L8.02009 8.68689L7.31299 7.97978L11.2928 4H8.33321V3Z' fill='%23646970'/%3E%3Cpath d='M6.33333 4.1665H4.33333C3.8731 4.1665 3.5 4.5396 3.5 4.99984V11.6665C3.5 12.1267 3.8731 12.4998 4.33333 12.4998H11C11.4602 12.4998 11.8333 12.1267 11.8333 11.6665V9.6665' stroke='%23646970'/%3E%3C/svg%3E%0A");				width: 24px;
+				width: 16px;
```
Comment (andfinally) [inline plugins/woocommerce/src/Internal/Admin/Marketplace.php]: Stray `width` rule here.

```suggestion
				background-image: url("data:image/svg+xml,%3Csvg width='16' height='16' viewBox='0 0 16 16' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill-rule='evenodd' clip-rule='evenodd' d='M8.33321 3H12.9999V7.66667H11.9999V4.70711L8.02009 8.68689L7.31299 7.97978L11.2928 4H8.33321V3Z' fill='%23646970'/%3E%3Cpath d='M6.33333 4.1665H4.33333C3.8731 4.1665 3.5 4.5396 3.5 4.99984V11.6665C3.5 12.1267 3.8731 12.4998 4.33333 12.4998H11C11.4602 12.4998 11.8333 12.1267 11.8333 11.6665V9.6665' stroke='%23646970'/%3E%3C/svg%3E%0A");
				width: 16px;
```

File: plugins/woocommerce/src/Internal/Admin/Marketplace.php [hunk 3]
```diff
@@ -12,13 +12,20 @@
...
 	 */
 	final public function init() {
 		if ( FeaturesUtil::feature_is_enabled( 'marketplace' ) ) {
 			add_action( 'admin_menu', array( $this, 'register_pages' ), 70 );
 			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+
+			// Add a Woo Marketplace link to the plugin install action links.
+			add_filter( 'install_plugins_tabs', array( $this, 'add_woo_plugin_install_action_link' ) );
+			add_action( 'install_plugins_pre_woo', array( $this, 'maybe_open_woo_tab' ) );
+			add_action( 'admin_print_styles-plugin-install.php', array( $this, 'add_plugins_page_styles' ) );
```
Comment (andfinally) [inline plugins/woocommerce/src/Internal/Admin/Marketplace.php:28]: Feels a bit icky inserting a style block, but I suppose it beats enqueuing a special stylesheet.

**Review by andfinally — APPROVED** [review]
Awesome, thanks Bero! 🙌

It might make sense to base this branch on `trunk` and do it separately from the My Subscriptions feature branch, as it's a more general Marketplace change. If we did to that, we'd have to create a changelog. But if it saves time to include it in the feature branch that's fine.

The external icon is not strictly applicable, as the link stays within the site. But it does have a useful function in warning people the link jumps out of the `plugin-install` context.

File: plugins/woocommerce/src/Internal/Admin/Marketplace.php (hunk 3 above)
Comment (bgrgicak) [reply plugins/woocommerce/src/Internal/Admin/Marketplace.php:28]: > Feels a bit icky inserting a style block, but I suppose it beats enqueuing a special stylesheet.

I didn't want to add another file download just for this.
If it grows I would move it to a file.

File: plugins/woocommerce/src/Internal/Admin/Marketplace.php (hunk 1 above)
Comment (bgrgicak) [reply plugins/woocommerce/src/Internal/Admin/Marketplace.php]: True, updated it now.

**Comment by bgrgicak** [comment]
> It might make sense to base this branch on trunk and do it separately from the My Subscriptions feature branch

If that's ok I would like to keep it in the feature branch to avoid conflicts now (we need to merge it)

**Review by Dan-Q — APPROVED** [review]
Works a treat.

**Pull request merged**